### PR TITLE
Add lookups

### LINKED ISSUE #41: Implement PLONKup
This issue will include the addition of lookups into the code via the plonkup proving system.

### PR DESCRIPTION
**Pull request by LukePearson1**
Resolves #41

### CODE REVIEW

File: src/lookup/lookup.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#![allow(clippy::too_many_arguments)]
+
+use crate::constraint_system::StandardComposer;
+use crate::constraint_system::Variable;
+use dusk_bls12_381::BlsScalar;
+
+impl StandardComposer {
+    /// Adds a plookup gate to the circuit with its corresponding 
+    /// constraints.
+    ///
```
Comment (stechu) [inline src/lookup/lookup.rs]: would you mind writing the gate equation explicitly here?

Comment (LukePearson1) [reply src/lookup/lookup.rs]: For now this has been removed @stechu. Only because lookups will be added in different PR's.

**Comment by LukePearson1** [comment]
When this PR is ready, I think we merge this and add lookups as a module and then organise the proving system in a follow up PR.

File: plonk-core/src/lookup/lookup_table.rs [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+    /// Insert a new row for an addition operation.
+    /// This function needs to know the upper bound of the amount of addition
+    /// operations that will be done in the plookup table.
+    pub fn insert_add_row(&mut self, a: u64, b: u64, upper_bound: u64) {
+        let c = (a + b) % upper_bound;
+        self.0.push([F::from(a), F::from(b), F::from(c), F::zero()]);
+    }
+
+    /// Insert a new row for an addition operation.
```
Comment (mathcrypto) [inline plonk-core/src/lookup/lookup_table.rs:42]: insert a row for a special operation not addition

**Comment by davidnevadoc** [comment]
Is this PR still relevant or is the work for PlookUp being carried out in the `add_lookups_with_refactor` branch?

**Comment by davidnevadoc** [comment]
Closing this stale PR. The implementation of PlonkUp is continued in PR #123

**Pull request closed without merging**